src/libraries/Common/src/System/Net/IPAddressParser.Common.cs

```diff
@@ -0,0 +1,47 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Numerics;
+
+namespace System.Net
+{
+    internal static partial class IPAddressParser<TChar>
+        where TChar : unmanaged, IBinaryInteger<TChar>
+    {
+        public const int Octal = 8;
+        public const int Decimal = 10;
+        public const int Hex = 16;
+
+        internal const int MaxIPv4StringLength = 15; // 4 numbers separated by 3 periods, with up to 3 digits per number
+        internal const int MaxIPv6StringLength = 65;
+
+        // Generic constants which are used for trying to parse a single digit as an integer.
+        private static readonly TChar NumericRangeStartCharacter = TChar.CreateTruncating('0');
```

Comment:
I don't think extracting constants like this is useful for readability. I'd stick to defining them inline (`TChar.CreateChecked('0')`).